src/webgpu/shader/validation/variable_and_const.spec.ts

```diff
@@ -103,3 +103,48 @@ g.test('v_0033')
       (lhsScalarType === rhsScalarType && lhsContainerType === rhsContainerType) || 'v-0033';
     t.expectCompileResult(expectation, code);
   });
+
+g.test('v_0038')
+  .desc(
+    `Tests for validation rule v-0038:
+  The following types are IO-shareable:
+  numeric scalar types
+  numeric vector types
+  Matrix Types
+  Array Types if its element type is IO-shareable, and the array is not runtime-sized
+  Structure Types if all its members are IO-shareable
+
+  As a result these are not IO-shareable:
+  boolean
+  vector of booleans
+  array of booleans
+  matrix of booleans
+  array runtime sized -> cannot be used outside of a struct, so no cts for this
+  struct with bool component
+  struct with runtime array
+  TODO: add test for structs`
+  )
+  .params(
+    params()
+      .combine([
+        { variableOrConstant: 'var', inputOrOutput: 'in' },
+        { variableOrConstant: 'var', inputOrOutput: 'out' },
```

Comment:
It would be ideal to have a "control case" to make sure that when *only* the storage class changes, the shader becomes invalid (to make sure nothing else is wrong).

Here I think the control case could be `var<private>` - all test cases should be valid (I think) with storage class private.